src/lib/containers/provenance/EntityNodeLabel.tsx

```diff
@@ -0,0 +1,48 @@
+import React, { useMemo } from 'react'
+import {
+  isVersionableEntityType,
+  convertToEntityType,
+} from '../../utils/functions/EntityTypeUtils'
+import { PRODUCTION_ENDPOINT_CONFIG } from '../../utils/functions/getEndpoint'
+import Tooltip from '../../utils/tooltip/Tooltip'
+import { ProvenanceEntityIcon } from './ProvenanceEntityIcon'
+import { EntityHeaderIsCurrent } from './ProvenanceUtils'
+
+export const EntityNodeLabel = (data: EntityHeaderIsCurrent) => {
+  const { entityHeader, isCurrentVersion } = data
+  const targetVersionNumberString = entityHeader.versionNumber
+    ? `.${entityHeader.versionNumber}`
+    : ''
+  const entityVersionString = `${entityHeader.id}${targetVersionNumberString}`
+  return useMemo(
+    () => (
+      <>
+        <div>
+          <ProvenanceEntityIcon entityHeader={entityHeader} />
+        </div>
+        <>
+          <div>
+            <Tooltip
+              title={entityHeader.name}
+              placement="top"
+              enterNextDelay={300}
+            >
+              <a
+                href={`${PRODUCTION_ENDPOINT_CONFIG.PORTAL}#!Synapse:${entityVersionString}`}
+              >
+                {entityHeader.name}
+              </a>
+            </Tooltip>
+          </div>
+          {isVersionableEntityType(convertToEntityType(entityHeader.type)) && (
+            <div>
+              <div>v.{entityHeader.versionNumber}</div>
+              {!isCurrentVersion && <div>(old version)</div>}
```

Comment:
That's a good point, the logic (for isCurrentVersion) and verbiage for tables needs to be updated.  This matches the current implementation of provenance.  Note that I am manually figuring out isCurrentVersion using EntityHeaders (not via the entity service), just like the current implementation, so it can be done in bulk.